### PR TITLE
SelectBox: Set "role" when fieldTemplate is rendered (T1230635)

### PR DESCRIPTION
**Pull request by jdvictoria**

### CODE REVIEW

File: packages/devextreme/testing/tests/DevExpress.ui.widgets.editors/selectBox.tests.js [hunk 1]
```diff
@@ -3326,9 +3326,37 @@ QUnit.module('search', moduleSetup, () => {
...
+    });
+
+    QUnit.test('component with fieldTemplate should have proper role attribute after search and selection (T1230696, T1230971)', function(assert) {
+        const $selectBox = $('#selectBox').dxSelectBox({
+            dataSource: ['one', 'two', 'three'],
+            fieldTemplate: () => {
+                return $('<div>').dxTextBox({});
+            },
+            searchEnabled: true,
+            itemTemplate: () => {
```
Comment (EugeniyKiyashko) [inline packages/devextreme/testing/tests/DevExpress.ui.widgets.editors/selectBox.tests.js]: We can remove this setting. It doesn't matter for testing scenario

File: packages/devextreme/testing/tests/DevExpress.ui.widgets.editors/selectBox.tests.js [hunk 2]
```diff
@@ -3326,9 +3326,37 @@ QUnit.module('search', moduleSetup, () => {
...
         assert.strictEqual($input.attr('aria-required'), 'true', 'aria-required should stay true after search and selection');
 
-        assert.strictEqual($input.attr('aria-haspopup'), 'listbox', 'initial render should have aria-haspopuphaspopup attribute set to listbox');
+        assert.strictEqual($input.attr('aria-haspopup'), 'listbox', 'aria-haspopup should stay to listbox after search and selection');
 
-        assert.strictEqual($input.attr('aria-autocomplete'), 'list', 'initial render should have aria-autocomplete attribute set to list');
+        assert.strictEqual($input.attr('aria-autocomplete'), 'list', 'aria-autocomplete should stay to list after search and selection');
+    });
+
+    QUnit.test('component with fieldTemplate should have proper role attribute after search and selection (T1230696, T1230971)', function(assert) {
```
Comment (EugeniyKiyashko) [inline packages/devextreme/testing/tests/DevExpress.ui.widgets.editors/selectBox.tests.js]: Set correct Ticket ID inside brackets

**Review by EugeniyKiyashko — DISMISSED** [review]

**Comment by jdvictoria** [comment]
[SelectBox - The input element has incorrect role attribute value if async templates are used [T1230635]](https://trello.com/c/Zobx2cfX/8812-selectbox-the-input-element-has-incorrect-role-attribute-value-if-async-templates-are-used-t1230635)

File: packages/devextreme/testing/tests/DevExpress.ui.widgets.editors/dropDownEditor.tests.js [hunk 3]
```diff
@@ -2169,6 +2169,33 @@ QUnit.module('aria accessibility', () => {
...
+
+        assert.strictEqual($input.attr('role'), 'combobox', 'role attribute should retain to combobox after typing');
+
+        keyboardMock($input)
+            .caret(1)
+            .press('backspace');
+
+        $input = $dropDownEditor.find(`.${TEXT_EDITOR_INPUT_CLASS}`);
+
+        assert.strictEqual($input.attr('role'), 'combobox', 'role attribute should retain to combobox after deleting');
```
Comment (AlexanderMoiseev) [inline packages/devextreme/testing/tests/DevExpress.ui.widgets.editors/dropDownEditor.tests.js:2197]: 'role attribute should remain assigned to the combobox after deleting'

Comment (AlexanderMoiseev) [inline packages/devextreme/testing/tests/DevExpress.ui.widgets.editors/dropDownEditor.tests.js]:
```suggestion
        assert.strictEqual($input.attr('role'), 'combobox', ''role attribute should remain assigned to the combobox after deleting'');
```

**Review by AlexanderMoiseev — DISMISSED** [review]

**Review by EugeniyKiyashko — APPROVED** [review]

**Review by AlexanderMoiseev — APPROVED** [review]

**Pull request merged**